src/basilisp/contrib/pytest/testrunner.py

```diff
@@ -269,6 +271,12 @@ def collect(self):
         filename = self.path.name
         module = self._import_module()
         ns = module.__basilisp_namespace__
+
+        # Ensure the test module was loaded because it was directly
+        # relative to an entry in `sys.path`.
+        if module.__name__ != munge(str(ns)):
```

Comment:
Does this cause problems with the changes introduced by #1176 ?

Reply:
Yes, and in particular it was breaking the `test_ns_not_in_syspath` test. 

Consider the following file relative to the project directory:

`./test/a/test_path.lpy`, declared with the namespace `a.test-path`.

Running basilisp test will locate and load `a.test-path` due to `""` in sys.path, but this is inconsistent because the namespace doesn't match the load path (it’s not `test.a.test-path`). Thus, an exception is thrown in this case.

(To make this work, the user must add the `-p test` option to the cli tool)